core/hmap.go

```diff
@@ -0,0 +1,39 @@
+package core
+
+import (
+	"errors"
+
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+type HMap = *orderedmap.OrderedMap[string, string]
+
+func hashMapBuilder(args []string, currentHashMap HMap) (HMap, int64, error) {
```

Comment:
Is passing `args[]` required here? Asking as we're any ways doing `iter := 1`
Instead of passing the entire args slice, shall we pass only pass key-value pairs?
With call as `keyValuePairs := args[1:]`.